### PR TITLE
Database dev v2

### PR DESCRIPTION
**Pull request by MaHDiaLaGaB**

### CODE REVIEW

File: conftest.py [hunk 1]
```diff
@@ -1,7 +1,41 @@
...
-post = {
-    'text': 'Everything you need to know to preorder Hogwarts Legacy, the upcoming open-world RPG set in '
-            'the wizarding world of Harry Potter, before the hero arrived.',
-    'photo': 'https://assets-prd.ignimgs.com/2022/03/18/hogwarts-legacy-4-1647622573681.jpeg?width=1280',
-    'time': datetime.datetime(2022, 3, 24, 16, 31, 9)}
+scraped_data = [{
+            'text': 'Welcome to the very first episode of Rogue Jam!\n\nOur judges are on the hunt for a game with '
+                    'Huge Potential, so join them as they tear through a Canadian wilderness rogue-like, '
+                    'a time-bending 2D shooter, and a hockey game with heart.',
+            'photo': 'http://s3.amazonaws.com/photos.sproutsocial.com/2022-04-14/2f1926ff-832a-4e26-81b7-ea8def6af7b4'
```
Comment (Moejoe90) [inline conftest.py:7]: Hmmm storing images in the form of a link is a problem I would say as the photo might get deleted

File: database.py [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+
+    def add_relationship(self, page_name, time):
+        with self.driver.session() as session:
+            try:
+                session.write_transaction(self._create_relationship, page_name, time)
+                print(f"{page_name}--[{time}]-->'post'")
+            except Exception(exceptions.SessionExpired):
+                raise
+
+    def fine_page(self, page_name):
```
Comment (Moejoe90) [inline database.py:119]: Find you mean here right?

File: database.py [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+    def check_connection(self):
+        self.driver.verify_connectivity()
+
+    def _create_page_node(self, tx, name):
+        try:
+            return tx.run("CREATE (a:Page {Name: $name, Created_at: $timestamp})", name=name, timestamp=ts)
+        except Exception(exceptions.CypherSyntaxError):
+            raise
+
+    def _create_relationship(self, tx, page_name, time):
```
Comment (Moejoe90) [inline database.py:58]: You're still using time only instead of an md5 hash

**Review by Moejoe90 — CHANGES_REQUESTED** [review]
Looks good we're almost there

File: database.py (hunk 2 above)
Comment (MaHDiaLaGaB) [reply database.py:119]: ops

File: test_database.py [hunk 4]
```diff
@@ -0,0 +1,40 @@
+from database import DataBase
```
Comment (Moejoe90) [inline test_database.py:1]: Very nice tests

File: database.py [hunk 5]
```diff
@@ -0,0 +1,178 @@
...
+        try:
+            return result
+        except ServiceUnavailable as exception:
+            logging.error(f"{query} raised an error: \n {exception}")
+            raise
+
+    def _create_post(self, tx, post_text, image, time):
+        hashed_dict = self._hash_everything(post_text, time)
+        hashed_text = hashed_dict['text'].hexdigest()
+        hashed_time = hashed_dict['time'].hexdigest()
```
Comment (Moejoe90) [inline database.py:83]: _hash_everything should do this 

Basically here you do

hash = self._hash_everything. 

Also no need to seperate text and time I'll tell you more in the method

File: database.py [hunk 6]
```diff
@@ -0,0 +1,178 @@
...
+            "RETURN b.Time AS time "
+            "Order by b.Time desc "
+            "Limit 1"
+        )
+        result = tx.run(query, page_name=page_name)
+        return [row['time'] for row in result]
+
+    def _hash_everything(self, text, time):
+        hashed = {'text': hashlib.md5(str(text).encode()),
+                  'time': hashlib.md5(str(time).encode())}
```
Comment (Moejoe90) [inline database.py:120]: it would make sense to return one hashed string here no need to seperate

**Review by Moejoe90 — CHANGES_REQUESTED** [review]
I left one request almost there :)

**Review by Moejoe90 — APPROVED** [review]

**Pull request merged**